syntax = "proto3";

package user_engagement;

enum EngagementAction {
  ENGAGEMENT_ACTION_UNSPECIFIED = 0;
  ACTION_NONE     = 1;  // basically home page / idle
  ACTION_DEBATING = 2;  // currently in a debate
}

// Action-specific payloads

message NoneInfo {
  // literally nothing
}

message DebatingInfo {
  string debate_id = 1;
  string current_claim_id = 2; // current claim being viewed/argued
}

// Main engagement message

message UserEngagement {
  string user_id = 1;
  EngagementAction current_action = 2;

  oneof action_info {
    NoneInfo     none_info     = 3;
    DebatingInfo debating_info = 4;
  }
}
